server/fileCompress/app.py

```diff
@@ -0,0 +1,103 @@
+import os
+import zipfile
+import json
+import posixpath
+from pprint import pprint
+import boto3
+import shutil
+
+
+def lambda_handler(event, context):
+    # Define the constants
+    TABLE_NAME = os.environ["JOB_TABLE"]
+    BUCKET_NAME = os.environ["SRC_BUCKET"]
+    PRIMARY_KEY = "id"
+    LOOKUP_BATCH_ID = event["queryStringParameters"]["id"]
+
+    # Create table and bucket reference
+    table = boto3.resource("dynamodb").Table(TABLE_NAME)
+    bucket = boto3.resource("s3").Bucket(BUCKET_NAME)
+
+    # Get the data from DynamoDB
+    response = table.get_item(Key={PRIMARY_KEY: LOOKUP_BATCH_ID})
+
+    # Download files using reference data from DynamoDB
+    files_json = response["Item"]["files"]
+
+    # Create zip file
+    zip_file_path = os.path.join("/tmp/", "temp_zip.zip")
```

Comment:
os.path.join adds the slashes for you, as they aren't the same on all operating systems. You probably want to be using Path instead, based on later lines that get the end of the path.

Reply:
Not sure what you mean by Path? For some reason, lamda does not like absolute paths to the tmp folder so I've replaced "/tmp/" with os.sep + "tmp" to solve this.

Reply:
The Path class from the standard pathlib module. i.e. ```from pathlib import Path```